.github/workflows/publish-release.yml

```diff
@@ -62,18 +79,29 @@ jobs:
     runs-on: ubuntu-latest
     needs: publish-npm-dry-run
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
         with:
           ref: ${{ github.sha }}
-      - uses: actions/cache@v3
+      - name: Install Node
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: '.nvmrc'
+      - name: Install Yarn
+        run: corepack enable
+      - name: Restore Yarn cache
```

Comment:
Hmm, this looks like a mistake. Why would we install Yarn here, or dependencies? This is in the step of publishing where we're trying to avoid running any code associated with the repository, to protect the publishing token from exfiltration in case our repo is compromised.

Reply:
Hmm, you're right. We don't need to install dependencies here. But I feel like we may need to install Yarn, because `action-npm-publish` expects it to be present. I believe this worked before without an explicit installation step because Yarn v1 is already present on GitHub runners and because the Yarn v3 binary was present in the repo. But now if we want to continue to use the same version of Yarn as other workflows we need to explicitly install v4. Does that make sense?

Reply:
Ah I see, yes that makes sense